src/Analysis/Abstract/Graph.hs

```diff
@@ -131,8 +132,9 @@ graphingModules recur m = do
       _             -> pure ()
   where
     -- NB: path is null for Languages like Ruby that have module imports that require concrete value semantics.
+    -- JZ:TODO: Shall we use Maybe File here?
     includeModule path
-      = let path' = if Prelude.null path then "unknown, concrete semantics required" else path
+      = let path' = if Prelude.null (Path.toString path) then Path.absRel "unknown, concrete semantics required" else path
```

Comment:
Shall we use Either or Maybe here?  It seems very strange for `Path.absRel "unknown, concrete semantics required"` here.

Reply:
Hmm, good question. Passing a non-path to `absRel` is definitely not the right thing to do; I'd probably go with the root directory, or `Path.emptyPath`.

Reply:
It seems we can put it as 
```
path' = path
```

In pathtype, the only way to make `Path.toString path == ""` is `path = Path.emptyFile`.

I make the change as the commit eb04077f1, we can revert this commit if you think it is right.